hedera-node/src/main/java/com/hedera/services/state/logic/RecordStreaming.java

```diff
@@ -75,9 +99,43 @@ public void run() {
 	}
 
 	public void stream(final RecordStreamObject rso) {
+		if (isInNewBlock(rso)) {
+			Hash computedHash = new Hash();
+			try {
+				computedHash = curRunningHashLeaf.getRunningHash().getFutureHash().get();
+			} catch (final InterruptedException e) {
+				log.error("Error in computing hash for block #{}", curNetworkCtx.getBlockNo());
+				Thread.currentThread().interrupt();
+				throw new RuntimeException(String.format("Failed to calculate hash for block #%d", curNetworkCtx.getBlockNo()));
+			}
```

Comment:
To be honest, I'm not sure how should we handle this scenario. What should we do if the future of a runningHashLeaf throws InterruptedException and we don't calculate the last running hash. If we continue the execution we will have a gap in the block hashes in the cache and won't be able to get this value back.

What do you think @tinker-michaelj ?

Reply:
In general, if a node has its `handleTransaction` thread interrupted, that's unrecoverable---it will almost certainly hit an ISS soon.

So imo just logging a very loud `ERROR` is about all we can do. 😞